src/transaction/guards.ts

```diff
@@ -35,9 +35,12 @@ export const isTxInput = (
 export const isTxOutput = (
   test: any,
 ): test is TxOutput => {
-  if (!Array.isArray(test) || test.length !== 2) return false
-  const [address, coins] = test
-  return Buffer.isBuffer(address) && isLovelace(coins)
+  if (Array.isArray(test) && test.length === 2) {
+    const [address, coinsOrMA] = test
```

Comment:
If you agree, we can rename the `coinsOrMa` to `amount` on all places.

Reply:
Refactored